tools/monitoring/cloudmail/mail.go

```diff
@@ -0,0 +1,197 @@
+/*
+Copyright 2019 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cloudmail
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	mail "cloud.google.com/go/mail/apiv1alpha3"
+	mailpb "google.golang.org/genproto/googleapis/cloud/mail/v1alpha3"
+	"google.golang.org/genproto/protobuf/field_mask"
+)
+
+const (
+	projectID = "knative-tests"
+	region    = "us-central1"
+
+	addressSetID   = "monitoring-address-set"
+	addressPattern = "monitoring-alert"
+	senderID       = "monitoring-alert-sender"
+	receiptRuleID  = "monitoring-receipt-drop-rule"
+)
+
+// CreateDomain creates a new project domain for cloud main
+func CreateDomain(ctx context.Context, client *mail.CloudMailClient) error {
+	domain := &mailpb.Domain{
+		ProjectDomain: true,
+		DomainName:    "",
+	}
+	req := &mailpb.CreateDomainRequest{
+		Parent: fmt.Sprintf("projects/%s", projectID),
+		Region: region,
+		Domain: domain,
+	}
+	resp, err := client.CreateDomain(ctx, req)
+	if err != nil {
+		return err
+	}
+	fmt.Printf("Domain created.\n")
+	fmt.Printf("Domain name: %s", resp.GetDomainName())
+	fmt.Printf("Domain region: %s", region)
+	fmt.Printf("Domain ID: %s", strings.Replace(resp.GetName(), fmt.Sprintf("regions/%s/domains/", region), "", 1))
```

Comment:
add the stuff we need from the reposnse to the struct. we should be holding onto these and not ask it from the caller?

Reply:
This is only called for once to create the domain. For the rest of the resource creation, it requires the domain id only; for sending the test mail, it requires the domain name. 

Updated it to print the info and return the domain name. Added a method to do the complete setup. Note: there's still value in having standalone setup-sender (without setting up the domain) since we only have one mail domain per project, but can have multiple senders.